inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/CommonConfigHolder.java

```diff
@@ -0,0 +1,477 @@
+                }
+                isInit = true;
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Get the original attribute map
+     *
+     * Notice: only the non-pre-read fields need to be searched from the attribute map,
+     *         the pre-read fields MUST be got according to the methods in the class.
+     */
+    public Map<String, String> getProperties() {
+        return this.props;
+    }
+
+    /**
+     * getStringFromContext
+     *
+     * @param context
+     * @param key
+     * @param defaultValue
+     * @return
+     */
+    public static String getStringFromContext(Context context, String key, String defaultValue) {
+        String value = context.getString(key);
+        value = (value != null) ? value : getInstance().getProperties().getOrDefault(key, defaultValue);
+        return value;
+    }
+
+    public String getClusterTag() {
+        return clusterTag;
+    }
+
+    public String getClusterName() {
+        return this.clusterName;
+    }
+
+    public String getClusterIncharges() {
+        return clusterIncharges;
+    }
+
+    public String getClusterExtTag() {
+        return clusterExtTag;
+    }
+
+    public long getConfigChkInvlMs() {
+        return configChkInvlMs;
+    }
+
+    public boolean isNoTopicAccept() {
+        return noTopicAccept;
+    }
+
+    public boolean isEnableWhiteList() {
+        return this.enableWhiteList;
+    }
+
+    public String getManagerType() {
+        return managerType;
+    }
+
+    public List<String> getManagerHosts() {
+        return this.ipListParser.getIpList();
+    }
+
+    public String getManagerAuthSecretId() {
+        return managerAuthSecretId;
+    }
+
+    public String getManagerAuthSecretKey() {
+        return managerAuthSecretKey;
+    }
+
+    public boolean isEnableAudit() {
+        return enableAudit;
+    }
+
+    public HashSet<String> getAuditProxys() {
+        return auditProxys;
+    }
+
+    public String getAuditFilePath() {
+        return auditFilePath;
+    }
+
+    public int getAuditMaxCacheRows() {
+        return auditMaxCacheRows;
+    }
+
+    public long getAuditFormatInvlMs() {
+        return auditFormatInvlMs;
+    }
+
+    public boolean isResponseAfterSave() {
+        return responseAfterSave;
+    }
+
+    public long getMaxResAfterSaveTimeout() {
+        return maxResAfterSaveTimeout;
+    }
+
+    public int getMaxBufferQueueSizeKb() {
+        return maxBufferQueueSizeKb;
+    }
+
+    public String getEventHandler() {
+        return eventHandler;
+    }
+
+    public String getCacheClusterSelector() {
+        return cacheClusterSelector;
+    }
+
+    public int getPrometheusHttpPort() {
+        return prometheusHttpPort;
+    }
+
+    public String getProxyNodeId() {
+        return proxyNodeId;
+    }
+
+    public String getMsgCompressType() {
+        return msgCompressType;
+    }
+
+    private void preReadFields() {
+        String tmpValue;
```

Comment:
Can we use Properties or JSON parsing tools to automatically parse the configuration in the file, and then map it to a Java POJO?
Now this kind of modification is not very flexible, such as modifying or adding a new configuration, you also need to modify the code.

Reply:
It is the construction of the Properties parsing mode, and the pre-reading fields are optimized for efficiency